plugins/gpio/fu-plugin-gpio.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+static void
+fu_plugin_gpio_init(FuPlugin *plugin)
+{
+	FuContext *ctx = fu_plugin_get_context(plugin);
+	fu_context_add_quirk_key(ctx, "GpioPrepare");
+	fu_context_add_quirk_key(ctx, "GpioCleanup");
+}
+
+static gboolean
+fu_plugin_gpio_create_node(FuPlugin *self, guint gpio_num, GError **error)
+{
+	g_autofree gchar *devname = g_strdup_printf("gpio%u", gpio_num);
+	g_autofree gchar *fn = g_build_filename("/sys/class/gpio", devname, NULL);
```

Comment:
why is this using the sysfs interface instead of the character device?  The character device should be a lot safer in a multi process system.

Reply:
....because I didn't know the sysfs api was deprecated until about 10 minutes ago :) Will fix.

Reply:
All fixed. It's a weird ioctl interface!